### PR TITLE
Kb cust valiation

### LINKED ISSUE #48: Auto-generate date when you create a customer
### Context
When you create a customer, it asks you to enter today's date. That's dumb! Let's auto-generate that date!

### Process
1. In ``createCustC.js``, autogenerate a date for date created
1. Remove the prompt for entering today's date in the ``createCustV.js``

### Expected Result
When you select option 1 on the main menu, you shouldn't have to enter today's date. When you go look at the new customer in the database, you the product should have today's date.

### LINKED ISSUE #51: Create new customer prompt has no restrictions on inputs
### Context
1. Creating a new customer has a bunch of prompts about name, address, etc.
1. Currently, the user can input whatever they like for each step of the prompt

### Process
1. Add `pattern` to each prompt in the `createCustV` and a regex forcing user input to match what we'd like it to

### Expected Result
1. Consistent entry of new customer data

### PR DESCRIPTION
**Pull request by kenziebottoms**
# Description
All customer creation prompts are validated now, and the date is auto generated.

## Related Ticket(s)
fixes #51, fixes #48

## Proposed Changes
- Names only take alphanumeric input
- Address and city only take alphanumeric and whitespace and dashes
- State only takes 2 capital letters
- Zip code only takes 5 integers
- Phone number only takes 10-11 digits with and without dashes, parentheses, and spaces

## Expected Behavior
You should get shit for trying to enter stuff other than the above for the prompts.

## Steps to Test Solution
1. Type gibberish into the prompts

## Testing

- [ ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
- [x] I certify that all existing tests pass

## Documentation
- [ ] There is new documentation in this pull request that must be reviewed..
- [ ] I added documentation for any new classes/methods

### CODE REVIEW

File: app/views/addCustPaymentOptV.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ module.exports.paymentOptionPrompts = [
     name: 'paymentType',
     description: 'Please enter a payment option',
     required: true,
-    type: 'string'
+    pattern: '^[0-9 ]*$'
```
Comment (DavidLarsKetch) [inline app/views/addCustPaymentOptV.js]: So this doesn't entirely restrict users to inputting to available payment type ids. We currently have payment types up to `18`. This regex would still allow a user to input `21` or `3214129` or whatever. We could throw in a `conform` here instead to make sure the payment type is restricted to the available ids.

**Review by DavidLarsKetch — DISMISSED** [review]
:+1: Works great!
See my above comment - not anything that needs to be addressed in this PR, just food for thought.

File: app/views/addCustPaymentOptV.js (hunk 1 above)
Comment (kenziebottoms) [reply app/views/addCustPaymentOptV.js]: #70

**Review by DavidLarsKetch — APPROVED** [review]
:+1:

**Review by joechesney — APPROVED** [review]
yups

**Pull request merged**